src/test/broadcastprofile_test.cpp

```diff
@@ -42,40 +42,39 @@ TEST(BroadcastProfileTest, SaveAndLoadXML) {
     // Preliminary: set a discriminating value in one of the profile fields
     QString streamName("unit testing in progress");
 
-    BroadcastProfile profile("Unit Testing Profile");
+    BroadcastProfile profile("Broadcast Profile test");
     profile.setStreamName(streamName);
 
-    QString filename = profile.getProfileName() + QString(".bcp.xml");
+    QTemporaryDir tempDir;
+    ASSERT_TRUE(tempDir.isValid());
+    QString filename = tempDir.filePath(profile.getProfileName() + QString(".bcp.xml"));
 
-    // Call save() on a profile and assert it actually exists
-    QFile::remove(filename); // First, make sure it doesn't exists
     profile.save(filename);
     ASSERT_TRUE(QFile::exists(filename));
 
     // Load XML file using static loadFromFile and assert
     // the discriminating value is present
     BroadcastProfilePtr savedProfile = BroadcastProfile::loadFromFile(filename);
-    ASSERT_NE(savedProfile, nullptr);
-    ASSERT_TRUE(savedProfile->getStreamName() == streamName);
+    EXPECT_NE(savedProfile, nullptr);
+    EXPECT_TRUE(savedProfile->getStreamName() == streamName);
```

Comment:
Reason for this change?

Reply:
Using EXPECT is correct here for finally validating the test result. ASSERT is mainly intended to be used while setting up the test.

https://github.com/google/googletest/blob/master/googletest/docs/primer.md#Assertions